### PR TITLE
Add TextSegmentationService. Integrate into TextClusteringService

### PR DESCRIPTION
**Pull request by annak13**
- Add url to configs
- Add TextSegmentationService to call external service
- Add TextSegmentationServiceTest
- Integrate into TextClusteringService in compute Embeddings

### Checklist
- [ ] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [X] ~Server: I added multiple integration tests (Spring) related to the features~
- [X] ~Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)~
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [X] ~Client: I added multiple integration tests (Jest) related to the features~
- [X] ~Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)~
- [X] ~Client: I documented the TypeScript code using JSDoc style.~
- [X] ~Client: I added multiple screenshots/screencasts of my UI changes~
- [X] ~Client: I translated all the newly inserted strings into German and English~

### Motivation and Context
This change improves the segmentation of answers to textual exercises into text blocks.

### Description

A new service is added that runs in a virtual machine in a docker container alongside with Text-Assessment-Clustering. The code is in the same repository.

### Steps for Testing

1. ~Run the code form the Segmentation folder of the Text-Assessment-Clustering repository the same way as the Clustering code~ (The Test Server is configured to use our VM for that. :+1:)
1. Log in to Artemis
2. Navigate to Overview
3. Chose a course and a text exercise with more than 10 submissions. Submissions should contain reasonable text on the same topic (as in a real exercise). Not Lorem Ipsum.
4. Change the due date for the exercise to trigger the calculations
5. Inspect the resulting text blocks.
    - In the Database, OR
    - When Assessing.

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/service/connectors/TextSegmentationServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+import de.tum.in.www1.artemis.domain.TextBlock;
+import de.tum.in.www1.artemis.domain.TextSubmission;
+import de.tum.in.www1.artemis.exception.NetworkingError;
+
+public class TextSegmentationServiceTest {
+
+    private static String SEGMENTATION_ENDPOINT = "http://localhost:8080/segment";
+
+    private static String EMBEDDING_ENDPOINT = "http://localhost:8080/segment";
```
Comment (annak13) [inline src/test/java/de/tum/in/www1/artemis/service/connectors/TextSegmentationServiceTest.java]: remove

**Comment by krusche** [comment]
@jpbernius what is the status of this PR? Seems to be outdated and has a lot of conflicting files now. Should we close this for now and reopen it when more progress is made?

**Comment by jpbernius** [comment]
This is related to ls1intum/Text-Assessment-Clustering#2.
There are no conflicts, but I am take care of this.

**Comment by krusche** [comment]
Ok, please make sure we have additional progress here until next Tuesday, otherwise I would prefer to close this PR and reopen it, when someone is actively working on it.
Otherwise I will lose the overview, if too many PRs are open

**Comment by jpbernius** [comment]
I did run into this problem when testing this now: `HttpMessageNotWritableException: Could not write JSON: Infinite recursion (StackOverflowError)`

```
Nov 26 09:27:00 vmbruegge105 java[26089]: Caused by: com.fasterxml.jackson.databind.JsonMappingException: Infinite recursion (StackOverflowError) (through reference chain: 
  de.tum.in.www1.artemis.domain.modeling.ModelingExercise["course"]
   ->de.tum.in.www1.artemis.domain.Course["exercises"]
  ->org.hibernate.collection.internal.PersistentSet[6]
  ->de.tum.in.www1.artemis.domain.modeling.ModelingExercise["tutorParticipations"]
  ->org.hibernate.collection.internal.PersistentSet[0]
   ->de.tum.in.www1.artemis.domain.TutorParticipation["trainedExampleSubmissions"]
  ->org.hibernate.collection.internal.PersistentSet[0]
  ->de.tum.in.www1.artemis.domain.ExampleSubmission["exercise"]
  ->de.tum.in.www1.artemis.domain.modeling.ModelingExercise["course"]
  ->...
```

I am investigating how `ModelingExercise` objects are loaded based on the Stacktrace:
```
  at org.springframework.web.client.RestTemplate.postForEntity(RestTemplate.java:445)
  at de.tum.in.www1.artemis.service.connectors.RemoteArtemisServiceConnector.invoke(RemoteArtemisServiceConnector.java:59)
  at de.tum.in.www1.artemis.service.connectors.RemoteArtemisServiceConnector.invokeWithRetry(RemoteArtemisServiceConnector.java:98)
  at de.tum.in.www1.artemis.service.connectors.TextSegmentationService.segmentSubmissions(TextSegmentationService.java:66)
  at de.tum.in.www1.artemis.service.connectors.TextSegmentationService.segmentSubmissions(TextSegmentationService.java:53)
  at de.tum.in.www1.artemis.service.TextClusteringService.getTextBlocks(TextClusteringService.java:162)
  at de.tum.in.www1.artemis.service.TextClusteringService.calculateClusters(TextClusteringService.java:102)
  at de.tum.in.www1.artemis.service.TextClusteringService$$FastClassBySpringCGLIB$$e05abc94.invoke(<generated>)
  at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
```

**Pull request merged**